OpenRA.Mods.Common/Traits/Upgrades/UpgradeManager.cs

```diff
@@ -17,12 +17,19 @@
 namespace OpenRA.Mods.Common.Traits
 {
 	[Desc("Attach this to a unit to enable dynamic upgrades by warheads, experience, crates, support powers, etc.")]
-	public class UpgradeManagerInfo : ITraitInfo, Requires<IUpgradableInfo>
+	public class UpgradeManagerInfo : ITraitInfo, IRulesetLoaded
 	{
 		public object Create(ActorInitializer init) { return new UpgradeManager(init); }
+		public void RulesetLoaded(Ruleset rules, ActorInfo info)
+		{
+			if (!info.Name.StartsWith("^") && !info.TraitInfos<IUpgradableInfo>().Any())
+				throw new Exception(
```

Comment:
Shall I make the same change in `TraitsInConstructOrder`? (Currently also throws Exception)